### ISSUE TITLE
Creating a particular schema with SQLite fails

### ISSUE THREAD

**Issue by Tobias4872** [typeorm/typeorm#231]
Hi,

I have been crafting some code at [https://github.com/Tobias4872/sqlite-typeorm-problem](https://github.com/Tobias4872/sqlite-typeorm-problem) which should help to replicated the problem.

The schema consists of a three level hierarchy (publisher -> site -> zone). The publisher itself owns instances of an entity called "player_specification" of which one is related to the zone (publisher -> player_specification <- zone).

I had some cascading setup to clear related entities on deletion, but have removed all of them except for the one which seems to be causing the problem. If a player specification is removed, I want to also remove all zones which are associated with this player specification, so I have added a ON DELETE CASCADE as follows:

```
    @ManyToOne((type) => PlayerSpecification, {
        onDelete: "CASCADE",
    })
    @JoinColumn()
    public playerSpecification: PlayerSpecification;
```

And this seems to be causing the problem, because when I remove the cascading, the schema is properly generated.

Any help on this would be appreciated.

Tobias.

**Comment by pleerock** [comment]
can you provide more info? Do you have error when you build a schema? Or schema is build, but its not correct? If its not correct then what exactly is incorrect?

**Comment by Tobias4872** [comment]
Hi,

did you see the reference to [this repo](https://github.com/Tobias4872/sqlite-typeorm-problem)? It describes the problem as well as the error I am getting.

Tobias.

**Comment by pleerock** [comment]
Thank you for this great example of reproduction. Ive found issue and trying to solve it.

**Comment by Luke265** [comment]
Here's the fix https://github.com/Luke265/typeorm/commit/7d74cd5d22ba288834e823dc19966a58e1d8dfdb

**Comment by pleerock** [comment]
This should be fixed in `0.0.8-alpha.2`

**Comment by Tobias4872** [comment]
Confirmed. Issues is fixed now. Thanks!!
